dashboard/app/models/sections/section.rb

```diff
@@ -109,10 +109,7 @@ def assign_code
     self.code = unused_random_code unless code
   end
 
-  #TODO(caleybrock) having this action on chaged causes a bug because a teacher might call
-  #disable when the default is already disable, and it will have no affect.
-  before_save :update_user_sharing, if: -> {sharing_disabled_changed?}
-  def update_user_sharing
```

Comment:
Note that now we have to call `update_student_sharing` explicitly, which is fine as it's handled in the API, but we lose the automatic model functionality.